firebase-app-distribution/src/main/java/com/google/firebase/app/distribution/NewReleaseFetcher.java

```diff
@@ -160,7 +160,15 @@ private boolean canInstall(AppDistributionReleaseInternal newRelease)
   private boolean isNewerBuildVersion(AppDistributionReleaseInternal newRelease)
       throws FirebaseAppDistributionException {
     return Long.parseLong(newRelease.getBuildVersion())
-        > getInstalledAppVersionCode(firebaseApp.getApplicationContext());
+            > getInstalledAppVersionCode(firebaseApp.getApplicationContext())
+        || !doesAppVersionNameMatch(newRelease);
```

Comment:
Didn't want to throw the below code in here since it didn't look great, but I am open to putting it back.